### DISCUSSION TITLE
Status: 400 BadRequest "message": "stream ended unexpectedly"

### DISCUSSION THREAD

**Discussion by 4konung** [adonisjs/core#1411] [Help]
I have an endpoint and controller to upload images:
```javascript
// /start/routes/index.ts
import Route from '@ioc:Adonis/Core/Route'

Route.group(() => {
  Route.post('/', 'ImagesController.upload').middleware('auth:api')
}).prefix('/api/images')

// /app/Controllers/Http/ImagesController.ts
import Env from '@ioc:Adonis/Core/Env'
import cloudinary from '@ioc:Services/Cloudinary'
import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
import { schema } from '@ioc:Adonis/Core/Validator'

export default class ImagesController {
  public async upload({
    request,
    response,
  }: HttpContextContract) {
    const { image } = await request.validate({
      schema: schema.create({
        image: schema.file({
          size: '50mb',
          extnames: ['jpg', 'png', 'jpeg'],
        }),
      }),
    })

    const result = await cloudinary.uploader.upload(image.tmpPath!, {
      tags: [`env-${Env.get('NODE_ENV')}`],
    })

    return response.status(200).send(result)
  }
}

```
nginx limit set to 50m (proxy-body-size: 50m)

When this runs locally, I can send with 'postman' any image (below size limit '50mb'), but when this code deployed in kuber, I've got this error even if I try to send 3Mb image.

Cloudinary doesn't even get called, because this error throws body-parser before service gets called. Any ideas or solutions to this problem?

**Comment by McSneaky** [comment]
Try to send image under 1mb, if I remember correctly by default it was 1 or 2mb upload size. Worth rechecking your K8 ingress config

**Reply by 4konung** [reply]
Actually it is 10m by default. 
We set limit at 50m at ingress settings, config/bodyparser.ts and in validator schema. And still got error response.

**Reply by McSneaky** [reply]
It usually happens when content-length header differs from actual stream size, or when some proxy cuts stream midway
Would require lil bit more background info to debug efficiently, but some things to look at
- Log out your ingress config to see if correct one is in use (maybe not applied?)
- Do you have any other LBs or proxies in between API and client (maybe some other proxy cuts it)?
- Check that you compiled TS to JS after config update, could also check that right value is in compiled JS
- Check FW rules, perhaps it's some firewall, which is blocking
- Tons of other networking settings that might break it
- Check ingress logs, most likely will get some hints from there, or from any other LB / proxy logs

**Reply by 4konung** [reply]
Thanks, will try to debug more with help of devopses.
